database/scrapers/faq_script.ts

```diff
@@ -67,124 +67,143 @@ async function uploadImage(imageUrl: string) {
   }
 
   const imageMetadata = (await fetchedImageMetadata.json()) as ImageMetadata;
-  const extension = imageMetadata.data.type.split("/")[1];
+  let extension: string | undefined = "";
+  if (imageMetadata.data.filename_disk !== null) {
+    if (imageMetadata.data.filename_disk.split(".").length > 1) {
+      extension = imageMetadata.data.filename_disk.split(".").pop();
+    }
+  }
+  if (extension === "") {
+    if (imageMetadata.data.filename_download !== null) {
+      if (imageMetadata.data.filename_download.split(".").length > 1) {
+        extension = imageMetadata.data.filename_download.split(".").pop();
+      }
+    }
+  }
+
+  if (extension === undefined || extension === "") {
+    throw new Error("Failed to determine image extension");
+  }
```

Comment:
the extension being undefined or empty is not a critical error, the `uploadStream` function will use a `.bin` extension in these cases instead.
you can log some message in this case if you want.